controllers/apps/transformer_cluster_deletion.go

```diff
@@ -120,15 +115,21 @@ func (t *clusterDeletionTransformer) Transform(ctx graph.TransformContext, dag *
 		}
 	}
 	delObjs = append(delObjs, toDeleteObjs(nonNamespacedObjs)...)
+
 	delKindMap := map[string]sets.Empty{}
 	for _, o := range delObjs {
 		// skip the objects owned by the component and InstanceSet controller
 		if shouldSkipObjOwnedByComp(o, *cluster) || isOwnedByInstanceSet(o) {
 			continue
 		}
+		// skip component objects since they are deleted in the previous step.
+		if _, ok := o.(*appsv1alpha1.Component); ok {
+			continue
+		}
```

Comment:
Will there be any non-component objects that the not-yet-deleted components rely on, but these non-component objects themselves have been deleted?

Reply:
It is possible, but the most impact would be an error fired during the reconciliation, without any substantial impact on the cluster.